DANS-KNAW/easy-deposit-api
# EASY-1554 unzip POSTed files

Fixes EASY-1554 unzip POSTed files

#### When applied it will
* unzip files when posted as sub-request of `Content-Type multipart/form-data`
* [x] have unit tests
* upload in a staging directory, only move to draft if
  * [x] no nested zip files are detected
  * [x] no zips between plain files
* [x] merge conflicts with #67 
* [ ] replace `addPayloadFile` with  `addStagedPayloadFile` when bag-lib is released with new method https://github.com/DANS-KNAW/easy-deposit-api/blob/02ea1237df1b82e79cd1301d8565d035de1c4f99/src/main/scala/nl.knaw.dans.easy.deposit/StagedFilesTarget.scala#L49

#### Where should the reviewer @DANS-KNAW/easy start?

#### How should this be manually tested?

See `readme.md` and `upload.html` in `src/test/resources/manual-test`

Try different formats (`.zip`, `.tar.gz`, a plain file with zip extension)  and character encodings. Examine the logging. Try a zip with and without a zip in it. Try to upload a zip with and without other files, if with: the zip in the first respectively in the second field.

#### Related pull requests on github
repo        /               PR                | note
------------------------------------------- | ----
https://github.com/DANS-KNAW/dans-bag-lib/pull/42    | Should provide a method to move a payload file into place from a staging area. Perhaps a method called `addStagedPayloadFile`? Not plural to keep the caller responsible for logic concerning overwriting or rejecting files. Note that in theory an uploaded file might be an existing directory or an uploaded directory an existing file.
